authcode/src/services/auth.ts

```diff
@@ -13,21 +13,21 @@ class AuthClient extends pxt.auth.AuthClient {
     protected onSignInFailed(): Promise<void> {
         return Promise.resolve();
     }
-    protected onUserProfileChanged(): Promise<void> {
-        const state = this.getState();
-        if (state.profile) {
+    protected async onUserProfileChanged(): Promise<void> {
```

Comment:
I had to mark the function `async`, which made an explicit return statement unnecessary in this case. async methods always return a promise, but the semantics are such that code can just return a value directly (e.g. `return true`) rather than a promise-resolved value (e.g. `return Promise.resolve(true)`). When an async method returns `Promise<void>`, an explicit return is unnecessary because it isn't returning a value and the promise is implicit.

Reply:
I always forget that fact that async functions automatically return a Promise, thanks!